package trackerclient;

import "lightning.proto";

message HelloToTracker {
    required string session_id = 1;
	required string address = 2;
	required int32 available = 3;
}

message HelloToClient {
	required string session_id = 1;
}

message GoodBye {
	required string session_id = 1;
}

message Ping {
	required int32 ping_data = 1;
}

message Pong {
	required int32 ping_data = 1;
}

message NewBatch {
	enum BatchType {
		RAYTRACE = 0;
		PHOTSHOOTAH = 1;
	}

	required string session_id = 1;
	required string batch_hash = 2;
	required int32 tasks = 3;
	required BatchType batch_type = 4;
}

message NodesRequest {
	required string session_id = 1;
	required int32 nodes = 2;
}

message NodesResponse {
	message Node {
		required string session_id  = 1;
		required string address = 2;
	}
	
	repeated Node nodes = 1;
}

